### PR TITLE
Turn Application into a PWA

### PR DESCRIPTION
**Pull request by caleb-mabry**
Allows for the URL to be "installable" onto devices. Just a base setup and can be expanded on later

### CODE REVIEW

File: webAO/sw.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+// Our service worker will intercept all fetch requests
+// and check if we have cached the file
+// if so it will serve the cached file
+self.addEventListener('fetch', (event) => {
+  event.respondWith(
+    caches.open(cacheName)
+      .then((cache) => cache.match(event.request, { ignoreSearch: true }))
+      .then((response) => response || fetch(event.request)),
+  );
+});
```
Comment (oldmud0) [inline webAO/sw.js:26]: do PWAs really require service workers? i just think service workers are botnets

**Review by oldmud0 — COMMENTED** [review]
what does this do in practice? is there a difference in how this app is presented on android

**Comment by caleb-mabry** [comment]
This should make the application "downloadable" in a sense. If you have an Android device I've implemented it for https://rippingresource.com

**Comment by stonedDiscord** [comment]
I tried it and it creates a shortcut that leads to chrome without a way to navigate away

**Comment by caleb-mabry** [comment]
Simply put, a PWA is a website with all the benefits of an app. PWAs give you a faster, more reliable, and more engaging version of your website or eCommerce store. PWAs can do most things that native apps can do, such as operate offline, access your camera and microphone if necessary, GPS, and more.

Kinda doesn't work without web. But this provides a better experience on phones by allowing you to "install" the app

File: webAO/sw.js (hunk 1 above)
Comment (caleb-mabry) [reply webAO/sw.js:26]: It's a requirement to be considered a PWA. Not sure why. This is pretty boilerplate code for getting it to work.

**Pull request merged**